File: 1518-water-bottles/1518-water-bottles.cpp
class Solution {
public:
    int numWaterBottles(int numBottles, int numExchange) {
        return numBottles + (numBottles - 1) / (numExchange - 1);
    }
};





// class Solution {
// public:
//     int numWaterBottles(int numBottles, int numExchange) {
//         int consumed = 0;
//         while (numBottles >= numExchange) {
//             consumed += numExchange;
//             numBottles -= numExchange;
//             numBottles += 1;
//         }
//         return consumed + numBottles;
//     }
// };

File: 1518-water-bottles/1518-water-bottles.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        return numBottles + (numBottles - 1) // (numExchange - 1)




# class Solution:
#     def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
#         consumed = 0
#         while numBottles >= numExchange:
#             consumed += numExchange
#             numBottles -= numExchange
#             numBottles += 1
#         return consumed + numBottles